### DISCUSSION TITLE
Web Development Discussion

### DISCUSSION THREAD

**Discussion by laurelhiatt** [dashnowlab/STRchive#36] [General]
I'm going to put a lightly hierarchical task list here in a moment

**Comment by laurelhiatt** [comment]
@vincerubinetti Hi!

**Reply by hdashnow** [reply]
@vincerubinetti I'm so pleased to hopefully be working with you! You should have write access to STRchive now. Let me know if you have any issues.

**Comment by laurelhiatt** [comment]
**Short term, lightly by priority**=
- [x] Search/filter/sort main table (DONE)
- [x] Hover on in-text citations for link
- [x] Citation section in-page
- [x] Organization of information in a way that is visually appealing (toggles/collapsible boxes?  different "views"?)
- [x] Locus-specific visualizations of main plots

**Maybe longer term, lightly by priority**
- [x] Automation of literature query (scheduled)
- [x] Automation of website update when database updates (automatic)
- [x] In-page visualization of data from other websites/databases

**Reply by laurelhiatt** [reply]
@hdashnow please let me know if I left off something major. @vincerubinetti , I will get you a napkin sketch maybe at the end of this week once you have the go-ahead to include us in your workload

**Reply by hdashnow** [reply]
I think plots should be higher on the list. Maybe second? Just because I think it will be more visually exciting. Otherwise, this looks great.

**Reply by laurelhiatt** [reply]
@vincerubinetti I really loved @avvaruakshay 's suggestion of a submission form on the website where someone can upload an annotated PDF with a note for us to review. has he already discussed this suggestion with you?

**Reply by avvaruakshay** [reply]
Hi @laurelhiatt No I haven't gotten in touch with @vincerubinetti . I'll discuss the ideas with him now.

**Reply by vincerubinetti** [reply]
Looks like we're going to talk about some design stuff tomorrow, but let me just comment on the uploading thing.

What I would suggest is just having users open an issue or discussion on GitHub where they attach the PDF or data, and you can review it from there.

If you really want to make it easier for users, say people without GitHub accounts, we could have a small "serverless" function running on Google Cloud that takes an anonymous form submission from the website and automatically creates an issue/discussion with the attachment. We did this for the monarchinitiative.org feedback form. In this case it'd be more complicated though because [GitHub doesn't have an _official_ way to do attachments through the API](https://stackoverflow.com/questions/45092776/is-it-possible-to-attach-a-file-to-a-github-issue-via-the-api).

If the goal was just to provide an easy upload for users, it seems like it'd be just as good to link to a public Dropbox or something where they could drag and drop a file.

If you're talking about having something more "official" solution, that makes the feature jump up a level in scope. We'd need a continuously running server and database to store user uploads. That would require bringing Faisal in, and also a more significant ongoing maintenance burden (hosting a static website is much simpler). For this project's size, I don't think this is warranted.

**Comment by laurelhiatt** [comment]
My concern is for folks without GitHub accounts who have never used it before, it may seem daunting to use the GitHub. Especially clinical researchers.

I think a public drop box would work though!

**Comment by hdashnow** [comment]
What about a form that sends any uploads to a shared email inbox or google drive in addition to creating an anonymous GitHub issue? I think it's okay if it requires the reviewer to take a second step to get the file, as long as it's getting saved somewhere easy to manage shared access.

**Reply by vincerubinetti** [reply]
I think that should be possible. To be clear, it'd look something like this:

The website has a form with an upload button, some text fields, whatever. On submission, it sends that data to our own "serverless" function on Google Cloud (just a cheap bit of server-side code that only runs when called, not continuously running). The server code then does the following. Uses Dropbox/Google Drive/whatever API (we'll need to register a key) to upload the file into a public folder. It returns a link to the file path upon creation. We then use the GitHub API to make a new anonymous issue (perhaps with the user's name, if we ask them to fill that out in the form), and in the body of the issue we paste the Dropbox link (this we can do easily, it's just attaching files that's not supported). Once all this is done, probably within a few seconds, the browser receives a response with the link to the created issue and Dropbox upload and we display it to the user.

**Reply by hdashnow** [reply]
Yep, that sounds awesome. We can also create a form/template for them to fill in so that we get stuff in a format that's useful.

**Reply by laurelhiatt** [reply]
Form with info that becomes a github issue: Name/Contact Email/New Locus or Locus Update?/Locus (if relevant)/Pertinent Details/Link to Paper or Paper DOI

**Comment by laurelhiatt** [comment]
https://fontawesome.com/icons/file-medical?f=classic&s=solid i like this for the medical icon

**Comment by laurelhiatt** [comment]
https://fontawesome.com/icons/database?f=classic&s=solid the database icon is a possibility for bioinformatic stuff?

**Comment by hdashnow** [comment]
What about DNA for the STR databases? https://fontawesome.com/icons/dna?f=classic&s=solid

**Comment by laurelhiatt** [comment]
Sure!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
